### PR TITLE
User consent

### PR DESCRIPTION
**Pull request by pyby**
### Motivation and Context

See #338.

### Description

- [x] Integrate `UserCentrics` core and UI SDKs.
- [x] Display first layer at first launch.
- [x] All to modify consent later, in settings, with the new Privacy section, and second layer.
- [x] Apply user contents for local SDK.
- [x] Share user consent to `SRGAnalytics` which has `TagCommander`.
- [x] Apply user contents for SRG Analytics (Comscore and TagCommander).
- [x] Apply user contents for SRG Letterbox (SRG Snitch).
- [x] Strategy when user consent update is displayed and player is playing.
- [x] Strategy when application is opened from deepking, notifications or User Activities, regarding modal opened or reset, and first banner displayed.

### Checklist

- The branch should be rebased onto the `develop` branch for whole tests with nighties, but it's not required.*
- The code followed the code style as `make quality` passes with a green tick on the last commit.
- [x] Remote configuration properties have been properly documented (if relevant).
- [x] The documentation has been updated (if relevant).
- [x] Issues are linked to the PR, if any.

\* The project uses Github merge queue feature, which rebases onto the `develop` branch before squashing and merging the PR.

### CODE REVIEW

File: Application/Sources/Application/Navigation.swift [hunk 1]
```diff
@@ -54,8 +54,18 @@ extension UIViewController {
...
+                }
+                .store(in: &cancellables)
+            
             let position = HistoryResumePlaybackPositionForMedia(media)
-            controller.playMedia(media, at: position, withPreferredSettings: nil)
+            controller.prepare(toPlay: media, at: position, withPreferredSettings: nil, completionHandler: {
+                if !UserConsentHelper.isShowingBanner {
+                    controller.play()
+                }
+            })
```
Comment (Loic-Dumas) [inline Application/Sources/Application/Navigation.swift:68]: You managed to layout superposition properly ?

File: Application/Sources/Application/UserConsentHelper.swift [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+        guard !hasRunSetup else { return }
+        
+        configureAndApplyConsents()
+        
+        hasRunSetup = true
+    }
+    
+    private static func configureAndApplyConsents() {
+        let options = UsercentricsOptions()
+        let ruleSetIdKey = ApplicationConfiguration.shared.isUserConsentCentralizedRuleSetPreferred ? "UserCentricsSRGRuleSetId" : "UserCentricsRuleSetId"
```
Comment (Loic-Dumas) [inline Application/Sources/Application/UserConsentHelper.swift:115]: ?

File: Application/Sources/Application/UserConsentHelper.swift [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+    }
+    
+    private static func printAcceptedServices(_ acceptedServices: [String]?) {
+        if let acceptedServices {
+            PlayLogDebug(category: "UserConsent", message: "Accepted templateIds:\n\(acceptedServices.joined(separator: "\n"))")
+        }
+        else {
+            PlayLogDebug(category: "UserConsent", message: "No consent has been given yet")
+        }
+    }
```
Comment (Loic-Dumas) [inline Application/Sources/Application/UserConsentHelper.swift:340]: I was planning to remove these debug content

Comment (Loic-Dumas) [reply Application/Sources/Application/UserConsentHelper.swift:340]: Maybe keep it ?

File: Application/Resources/Apps/Play RSI/it.lproj/Localizable.strings [hunk 4]
```diff
@@ -83,6 +83,9 @@
 /* Search setting option */
 "All" = "Tutto";
 
+/* Always ask user consent at launch setting label */
+"Always ask user consent at launch" = "Richiedere sempre il consenso dell'utente al momento dell'avvio";
+
```
[the remark this answers is not in the document]
Comment (pyby) [reply Application/Resources/Apps/Play RSI/it.lproj/Localizable.strings:88]: Should be kept on Apple and Android on debug/beta builds only.

File: Application/Resources/Apps/Play RTR/RTRResources.xcassets/Images/logo_bu_rtr.imageset/Contents.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    {
+      "filename" : "BU=RTR 2.pdf",
+      "idiom" : "tv"
+    }
+  ],
+  "info" : {
+    "author" : "xcode",
+    "version" : 1
+  }
+}
```
[the remark this answers is not in the document]
Comment (pyby) [reply Application/Resources/Apps/Play RTR/RTRResources.xcassets/Images/logo_bu_rtr.imageset/Contents.json:20]: BU logos used in  first banner.

File: Application/Sources/Application/UserConsentHelper.swift (hunk 3 above)
Comment (pyby) [reply Application/Sources/Application/UserConsentHelper.swift:340]: We can keep it.

File: Application/Sources/Application/UserConsentHelper.swift (hunk 2 above)
Comment (pyby) [reply Application/Sources/Application/UserConsentHelper.swift:115]: It was during tests, with BU rule set id or the main SRG one.
Test removed.

File: Application/Sources/Application/Navigation.swift (hunk 1 above)
Comment (pyby) [reply Application/Sources/Application/Navigation.swift:68]: Exact. The ` UserConsentHelper` has two methods to retain and release the banner to be displayed instantly.

Here, it's when the player had preload the content, start playing only if a banner is not displayed. Deeplink to a media could open the player page, then the first banner, then the player finishs to preload the content and ready to play, check it.

**Pull request merged**